src/seth/libexec/seth/seth-calldata

```diff
@@ -1,43 +1,36 @@
-#!/usr/bin/env node
-const usage = `Usage: seth calldata <sig> [<args>]>
-   or: seth calldata <file>
-   or: seth calldata <data>`;
-if (process.argv.length >= 3 &&
-    process.argv[2].indexOf('(') >= 0) {
-  // silence warnings we don't care about
-  const log = console.log
-  console.log = () => {};
-  const ethers = require("./ethers.min.js");
-  console.log = log;
-  const sig = process.argv[2];
-  try {
-    const funcs = new ethers.utils.Interface(["function " + sig.replace(')(', ') returns (')]).functions;
-    const args = process.argv.slice(3).map(s => {
-      try {
-        return JSON.parse(s)
-      }
-      catch (e) {
-        return s
-      }
-    })
-    console.log(funcs[Object.keys(funcs)[0]].encode(args))
-  } catch (e) {
-    console.error(e.toString())
-    process.exit(1)
-  }
-} else if (process.argv.length == 3) {
-  const {execFileSync}   = require('child_process');
-  try {
-    const yes = execFileSync("seth",
-                             ["--to-hexdata",
-                              process.argv[2]]
-                            ).toString().replace('\n','')
-    console.log(yes)
-  } catch(e) {
-    console.error(e)
-    process.exit(1)
-  }
-} else {
-  console.error(usage)
-  process.exit(1)
-}
+#!/usr/bin/env bash
+### seth-calldata -- pack a signature and a list of arguments into calldata
+### Usage: seth calldata <sig> [<args>]
+###    or: seth calldata <file>
+###    or: seth calldata <data>
+###
+### Pack <sig> and <args> into data suitable for passing to `seth call'.
+### If <sig> has the form `<name>(<types>)', pack <sig>/<args> into calldata.
+###
+### If <file> contains a slash character, treat it as a file name to read.
+### Otherwise, ensure <data> begins with `0x' and treat it as hexdata.
+set -e
+if [[ $1 =~ ^([^\(]+)\( ]]; then
+  name=${BASH_REMATCH[1]}
+  abi=$(seth --abi-function-json "$1" | jq .[0])
+  shift
+  for arg; do
+    [[ $arg = @* ]] && arg=$(seth lookup "$arg")
+    args+=" --arg \"${arg//\"/\\\"}\" "
+  done
+  abi=$(echo "${abi//\"/\\\"}" | tr -d '[:space:]')
+  set -x
+  calldata=$(
+    eval "hevm abiencode --abi \"${abi}\" --name $name $args"
```

Comment:
Do we really need `eval` here? Can't we just capture the output of the call?

Reply:
I was not able to preserve the necessary quotes around `abi` and the contents of `args` when doing a direct invocation

Reply:
Also, do you really want `set -x` in there?

Reply:
`calldata=$(hevm abiencode --abi "${abi}" --name $name $args` doesn't work?

Reply:
Anyway, I'm a bit against `eval` but if you don't wanna spend too much time finding another way, I'm ok with that.

Reply:
> `calldata=$(hevm abiencode --abi "${abi}" --name "$name" "$args"` doesn't work?

Not directly. There is probably a better way to do this, but I'm not proficient enough in bash to know it.

removed `set -x`